4.0/fr/0x04-Assessment_and_Certification.md

```diff
@@ -0,0 +1,59 @@
+
+Il n'est pas possible d'effectuer une vérification complète de l'ASVS en utilisant uniquement des outils de test de pénétration automatisés. Alors qu'une grande majorité des exigences de la L1 peuvent être réalisées à l'aide de tests automatisés, la majorité globale des exigences ne se prête pas à l'utilisation de tests d'intrusion automatisés.
+
+Veuillez noter que les limites entre les tests automatisés et manuels se sont estompées au fur et à mesure que le secteur de la sécurité des applications a évolué. Les outils automatisés sont souvent réglés manuellement par des experts et les testeurs manuels utilisent souvent une grande variété d'outils automatisés.
+
+#### Le rôle des tests de pénétration
+
+Dans la version 4.0, nous avons décidé de rendre la L1 entièrement testable par pénétration sans accès au code source, à la documentation ou aux développeurs. Deux éléments d'enregistrement, qui doivent être conformes au Top 10 2017 A10 de l'OWASP, nécessiteront des entretiens, des captures d'écran ou d'autres collectes de preuves, tout comme dans le Top 10 2017 de l'OWASP. Cependant, les tests sans accès aux informations nécessaires ne constituent pas une méthode idéale de vérification de la sécurité, car ils laissent de côté la possibilité d'examiner la source, d'identifier les menaces et les contrôles manquants, et de réaliser un test beaucoup plus approfondi dans un délai plus court. 
+
+Dans la mesure du possible, l'accès aux développeurs, à la documentation, au code et l'accès à une application de test avec des données de non-production, est nécessaire lors de l'exécution d'une évaluation de niveau 2 ou 3. Les tests de pénétration effectués à ces niveaux nécessitent ce niveau d'accès, que nous appelons "examens hybrides" ou "tests de pénétration hybrides". 
+
+## Autres utilisations de l'ASVS
+
+En plus de servir à évaluer la sécurité d'une application, nous avons identifié un certain nombre d'autres utilisations potentielles de l'ASVS.
+
+### Guide pour les tests unitaires et d'intégration automatisés
+
+L'ASVS est conçu pour être hautement testable, à la seule exception des exigences en matière d'architecture et de code malveillant. #FIXME# Grâce à des tests unitaires de construction et d'intégration qui permettent de détecter des cas de fuzz et d'abus spécifiques et pertinents, l'application se vérifie pratiquement d'elle-même à chaque construction. Par exemple, des tests supplémentaires peuvent être élaborés pour la suite de tests pour un contrôleur de connexion, en testant le paramètre de nom d'utilisateur pour les noms d'utilisateur communs par défaut, l'énumération des comptes, le forçage de brute, l'injection LDAP et SQL, et XSS. De même, un test sur le paramètre du mot de passe devrait inclure les mots de passe communs, la longueur du mot de passe, l'injection d'octets nuls, la suppression du paramètre, XSS, et plus encore.
```

Comment:
The ASVS is designed to be highly testable, with the sole exception of architectural and malicious code
requirements. By building unit and integration tests that test for specific and relevant fuzz and abuse cases, the
application becomes nearly self-verifying with each and every build. For example, additional tests can be crafted
for the test suite for a login controller, testing the username parameter for common default usernames, account
enumeration, brute forcing, LDAP and SQL injection, and XSS. Similarly, a test on the password parameter should
include common passwords, password length, null byte injection, removing the parameter, XSS, and more.